src/core/vms.rs

```diff
@@ -47,8 +50,15 @@ impl ResourceTrait for Vm {
     }
 
     fn price_per_hour(&self) -> Result<f32, ResourceError> {
+        // formula to calculate dedicated instance (https://docs.outscale.com/en/userguide/Getting-the-Price-of-Your-Resources.html)
         match self.price_per_hour {
-            Some(price) => Ok(price),
+            Some(price) => match self.tenancy.as_str() {
+                "dedicated" => {
+                    Ok(price + ((price * self.price_dedicated_vm_additional_per_hour) / 100.0))
+                }
+                "default" => Ok(price),
```

Comment:
This should be done in compute, not in price_per_hour, otherwise, the price_per_month is wrong

Reply:
Done